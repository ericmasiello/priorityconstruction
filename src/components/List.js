import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import * as CustomPropTypes from '../propTypes';

export const List = ({ tag: Tag, decorated, ...rest }) => <Tag {...rest} />;

List.propTypes = {
  tag: CustomPropTypes.Tag,
  decorated: PropTypes.bool,
};

List.defaultProps = {
  tag: 'ul',
  decorated: false,
};

List.displayName = 'List';

const StyleList = styled(List)`
  margin: 0;
  padding-left: 1.5rem;

  ${({ decorated }) =>
    !decorated &&
    `
    list-style-type: none;
    padding-left: 0;
  `};
`;

StyleList.propTypes = List.propTypes;
StyleList.defaultProps = List.defaultProps;

export default StyleList;
