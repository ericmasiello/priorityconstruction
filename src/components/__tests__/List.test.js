import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import List, { List as RawList, ListItem as RawListItem } from '../List';

it('should render <List>', () => {
  const component = renderer.create(<List />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render children', () => {
  const wrapper = shallow(<RawList>Test</RawList>);

  expect(wrapper.text()).toBe('Test');
});

it('should render additional props', () => {
  const wrapper = shallow(<RawList data-test="my-attribute" />);

  expect(wrapper.find('[data-test="my-attribute"]')).toHaveLength(1);
});

it('should display additional classes', () => {
  const wrapper = shallow(<RawList className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render as a <ul> by default', () => {
  const wrapper = shallow(<RawList />);

  expect(wrapper.type()).toBe('ul');
});

it('should render with a custom tag', () => {
  const wrapper = shallow(<RawList tag="span" />);

  expect(wrapper.type()).toBe('span');
});

it('should render <List.Item>', () => {
  const component = renderer.create(<List.Item />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render children', () => {
  const wrapper = shallow(<RawListItem>Test</RawListItem>);

  expect(wrapper.text()).toBe('Test');
});

it('should render additional props', () => {
  const wrapper = shallow(<RawListItem data-test="my-attribute" />);

  expect(wrapper.find('[data-test="my-attribute"]')).toHaveLength(1);
});

it('should display additional classes', () => {
  const wrapper = shallow(<RawListItem className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render as a <li> by default', () => {
  const wrapper = shallow(<RawListItem />);

  expect(wrapper.type()).toBe('li');
});

it('should render with a custom tag', () => {
  const wrapper = shallow(<RawListItem tag="span" />);

  expect(wrapper.type()).toBe('span');
});
