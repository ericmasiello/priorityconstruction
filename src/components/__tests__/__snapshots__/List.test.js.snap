// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <List> 1`] = `
.c0 {
  margin: 0;
  padding-left: 1.5rem;
  list-style-type: none;
  padding-left: 0;
}

<ul
  className="c0"
/>
`;

exports[`should render a decorated <List> 1`] = `
.c0 {
  margin: 0;
  padding-left: 1.5rem;
}

<ul
  className="c0"
/>
`;
